pkg/sub_controller/feproxy/feproxy_configmap.go

```diff
@@ -0,0 +1,131 @@
+package feproxy
+
+import (
+	"context"
+	"fmt"
+
+	srapi "github.com/StarRocks/starrocks-kubernetes-operator/pkg/apis/starrocks/v1"
+	rutils "github.com/StarRocks/starrocks-kubernetes-operator/pkg/common/resource_utils"
+	"github.com/StarRocks/starrocks-kubernetes-operator/pkg/k8sutils"
+	"github.com/StarRocks/starrocks-kubernetes-operator/pkg/k8sutils/load"
+	"github.com/StarRocks/starrocks-kubernetes-operator/pkg/k8sutils/templates/service"
+	"github.com/StarRocks/starrocks-kubernetes-operator/pkg/sub_controller/fe"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+)
+
+func (controller *FeProxyController) SyncConfigMap(ctx context.Context, src *srapi.StarRocksCluster) error {
+	feProxySpec := src.Spec.StarRocksFeProxySpec
+
+	feSpec := src.Spec.StarRocksFeSpec
+	config, err := fe.GetFeConfig(ctx, controller.k8sClient, &feSpec.ConfigMapInfo, src.Namespace)
+	if err != nil {
+		klog.Errorf("FeProxyController Sync: get fe configmap failed, "+
+			"namespace = %v, configmapName = %v, configmapKey = %v, error = %v",
+			src.Namespace, feSpec.ConfigMapInfo.ConfigMapName, feSpec.ConfigMapInfo.ResolveKey, err)
+		return err
+	}
+	httpPort := rutils.GetPort(config, rutils.HTTP_PORT)
+
+	feServiceName := service.ExternalServiceName(src.Name, src.Spec.StarRocksFeSpec)
+	proxyPass := fmt.Sprintf("http://%s:%d", feServiceName, httpPort)
+
+	resolver := feProxySpec.Resolver
+	if resolver == "" {
+		resolver = "kube-dns.kube-system.svc.cluster.local"
```

Comment:
whether we can call api server to get the accurate core-dns service? e.g. the core-dns deployed in a different namespace, or the kubernetes has a different domain name other than `.cluster.local`

Reply:
I do not think api-server knows that, and that is why I add a resolver filed in CRD